aptos-move/framework/aptos-token-objects/sources/collection.move

```diff
@@ -97,16 +102,50 @@ module aptos_token_objects::collection {
         mint_events: event::EventHandle<MintEvent>,
     }
 
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed supply tracker, this is useful for ensuring that a limited number of tokens are minted.
+    /// and adding events and supply tracking to a collection.
+    struct ConcurrentFixedSupply has key {
```

Comment:
ConcurrentFixedSupply and ConcurrentFixedSupply are the same, why we need both? We can keep one only, and the second `if` in increment can go away? If we want consistent errors having a boolean flag might be cleaner?

Reply:
So, we can have only one struct that tracks supply, and have an additional flag in the struct that says whether it is a fixed/unlimited supply?